src/main/scala/com/touk/hiring_task/multiplex/repositories/CinemaRepository.scala

```diff
@@ -0,0 +1,60 @@
+package com.touk.hiring_task.multiplex.repositories
+
+import com.touk.hiring_task.multiplex.commons.{MultiplexException, SelectedPlaceUnavailableException}
+import com.touk.hiring_task.multiplex.dto.OrderDto
+import com.touk.hiring_task.multiplex.model._
+import com.touk.hiring_task.multiplex.utils.SystemClock
+
+import java.util
+import java.util.Enumeration
+import java.util.concurrent.ConcurrentHashMap
+import scala.concurrent.{ExecutionContext, Future}
+
+trait CinemaRepository {
+  def fetchProjectionsByDate(startAt: Long, finishAt: Long): Future[Option[List[Projection]]]
+
+  def takeProjectionById(id: String): Future[Option[Projection]]
+
+  def fetchKeysDB(): Future[Option[Enumeration[Key]]]
+
+  def storeReservations(o: OrderDto, expire: Long, roomName: String): Future[Either[MultiplexException, Unit]]
+}
+
+class CinemaRepositoryImpl(fillDB: FillCinemaRoomsService)(implicit ec: ExecutionContext) extends CinemaRepository with SystemClock {
```

Comment:
the Impl suffix in child classes names almost always means that sth is wrong (sometimes only we would need a better name, but sometimes it may show us that the interface/trait maybe is not needed). Do you have an idea, how we could call this class?